src/server/network.ts

```diff
@@ -287,12 +305,26 @@ export class Response extends SdkObject {
     for (const { name, value } of this._headers)
       this._headersMap.set(name.toLowerCase(), value);
     this._getResponseBodyCallback = getResponseBodyCallback;
+    this._serverIPAddressAndPortPromise = new Promise(f => {
+      this._serverIPAddressAndPortPromiseCallback = f;
+    });
+    this._securityDetailsPromise = new Promise(f => {
+      this._securityDetailsPromiseCallback = f;
+    });
     this._finishedPromise = new Promise(f => {
       this._finishedPromiseCallback = f;
     });
     this._request._setResponse(this);
   }
 
+  _serverIPAddressAndPortFinished(serverIPAddress?: IPAddressAndPort) {
+    this._serverIPAddressAndPortPromiseCallback(helper.filterEmpties(serverIPAddress));
```

Comment:
I don't think you need filterEmpties here and below. Everything is eventually going to be JSON.stringified into the har.

Reply:
I've updated this, but we will get some empty objects for securityDetails in the HAR. (Tests updated accordingly.)